### PR TITLE
[facebook] Improve extractor

### LINKED ISSUE #3496: [Facebook] New URL doesn't work
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.04.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Indonesia / Singapore

### Description

There are new Facebook post URL: https://www.facebook.com/story.php?story_fbid=pfbid0Fnzhm8UuzjBYpPMNFzaSpFE9UmLdU4fJN8qTANi1Dmtj5q7DNrL5NERXfsAzDEV7l&id=100073071055552

The error only happens when I use --cookies, without --cookies it works fine.

### Verbose log

```shell
[debug] Command-line config: ['https://www.facebook.com/story.php?story_fbid=pfbid0Fnzhm8UuzjBYpPMNFzaSpFE9UmLdU4fJN8qTANi1Dmtj5q7DNrL5NERXfsAzDEV7l&id=100073071055552', '-vU', '--cookies', 'facebook.com_cookies.txt']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.04.08 [7884ade65]
[debug] Python version 3.8.6 (CPython 64bit) - macOS-10.16-x86_64-i386-64bit
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0 (setts), ffprobe 5.0, rtmpdump 2.4
[debug] Optional libraries: brotli, certifi, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2022.04.08, Current version: 2022.04.08
yt-dlp is up to date (2022.04.08)
[debug] [generic] Extracting URL: https://www.facebook.com/story.php?story_fbid=pfbid0Fnzhm8UuzjBYpPMNFzaSpFE9UmLdU4fJN8qTANi1Dmtj5q7DNrL5NERXfsAzDEV7l&id=100073071055552
[generic] story: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] story: Downloading webpage
[generic] story: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.facebook.com/story.php?story_fbid=pfbid0Fnzhm8UuzjBYpPMNFzaSpFE9UmLdU4fJN8qTANi1Dmtj5q7DNrL5NERXfsAzDEV7l&id=100073071055552
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/yt_dlp/YoutubeDL.py", line 1383, in wrapper
    return func(self, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/yt_dlp/YoutubeDL.py", line 1453, in __extract_info
    ie_result = ie.extract(url)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/yt_dlp/extractor/common.py", line 633, in extract
    ie_result = self._real_extract(url)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/yt_dlp/extractor/generic.py", line 4028, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.facebook.com/story.php?story_fbid=pfbid0Fnzhm8UuzjBYpPMNFzaSpFE9UmLdU4fJN8qTANi1Dmtj5q7DNrL5NERXfsAzDEV7l&id=100073071055552
```

### PR DESCRIPTION
**Pull request by hatienl0i2612**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Close #3496

### CODE REVIEW

File: yt_dlp/extractor/facebook.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ class FacebookIE(InfoExtractor):
                                 video/embed|
                                 story\.php|
                                 watch(?:/live)?/?
-                            )\?(?:.*?)(?:v|video_id|story_fbid)=|
+                            )\?(?:.*?)(?:v|video_id|story_fbid(.*?)?)=|
```
Comment (dirkf) [inline yt_dlp/extractor/facebook.py:50]: Isn't `(?:...|story_fbid(.*?)?)=` actually `(?:...|story_fbid.*?)=`?

**Comment by pukkandan** [comment]
This is wrong as far as I can tell. Looks like you should be matching `id=` instead

**Comment by Wikidepia** [comment]
I'm planning to fix this, should I create new PR?

File: yt_dlp/extractor/facebook.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ class FacebookIE(InfoExtractor):
                                 video/embed|
                                 story\.php|
                                 watch(?:/live)?/?
-                            )\?(?:.*?)(?:v|video_id|story_fbid)=|
+                            )\?(?:.*?)(?:v|video_id|story_fbid.*?)=|
```
Comment (dirkf) [inline yt_dlp/extractor/facebook.py:50]: So is this what is being proposed?
```suggestion
                            )\?(?:.*?)(?:v|video_id|id)=|
```
Actually, we could be stricter:
```suggestion
                            )\?(?:.*?&)(?:v|video_id|id)=|
```

**Comment by Lesmiscore** [comment]
> I'm planning to fix this, should I create new PR?

Yes.

**Pull request closed without merging**